### ISSUE TITLE
Импортировать время точки из gpx

### ISSUE THREAD

**Issue by yanenok** [wladich/nakarte#573]
Часто пользуюсь импортом gpx с последующим экспортом, чтобы редактировать существующий трек. При экспорте gpx в элемент time проставляется заглушка 1 января 1970 (#91), хотя в импортируемом треке он может быть проставлен. Было бы удобно импортировать время (и, возможно, другие метаданные) вместе с координатами и использовать его при экспорте.

**Comment by slazav** [comment]
Да, времени и высоты очень не хватает. В результате, редактирование записанных треков - та операция, которую не сделаешь в nakarte.

**Comment by wladich** [comment]
Для контекста, текущая ситуация такая: при чтении треков из любых форматов берутся только координаты, все остальные измерения игнорируются.  Дата в виде 1 января 1970 ставится для совместимости с Garmin Connect: https://github.com/wladich/nakarte/issues/90.
Почему так было сделано - я предполагал использовать сайт только для планирования пеших маршрутов, в т.ч. с использованием записанных треков; в этом случае все данные кроме координат не сильно важны. Кроме того, часть точек может быть удалена (расположенные ближе 2.4 м друг от друга и удаленные менее чем на 2.4 м от прямой, соединяющей соседние точки) -  для треков с особо большим количеством точек это улучшает отзывчивость интерфейса и упрощает редактирование.  А ещё при сохранении в ссылку координаты округляются до ближайшего значения кратного 2.4 м - это сделано для экономии места.

Вопрос про сохранение времени и высот периодически возникает, но я не знаю, как вписать это в имеющуюся модель и интерфейс. 

1. При загрузке / сохранении координаты точек (да и количество точек) меняются, не сильно и незаметно для пользователя, но, для задачи именно обработки записанных треков, мне кажется, это недопустимо.
2. После загрузки трека пользователь может не только разрезать треки и удалять точки, но и передвигать точки и добавлять новые точки и сегменты.  Для этих новых и измененных точек у нас нет дополнительных метрик (высота, время, пульс и т.п.), интерполировать в большинстве случаев невозможно. В итоге мы получим в одном треке часть точек без этих дополнительных метрик. Пользователь и сторонние программы могут быть сильно удивлены результатом.
3. Очевидное решение для п. 2 - при попытке редактирования загруженного трека, не связанной с удалением точек, показывать предупреждение, что "Сейчас вы потеряете высоту/время/температуру в треке" мне категорически не нравится - функция нужна явно меньше чем в 50% случаев и это сообщение будет раздражать.
4. Менее очевидное решение - разделить сущности на 1) планируемый маршрут и 2) записанный трек. Это приведёт к усложнению интерфейса.

В общем, хороших решений у меня для этой задачи пока нет.

**Comment by vsespb** [comment]
При импортировании .gpx в locus 4, треку проставляется дата 01.01.1970 и он попадает в конец списка треков в папке, у остальных треков в папке нормальная дата - то когда его создали или когда его записали (проехали). А эти 1970 выбиваются из этой стройной системы.
Если посмотреть содержимое .gpx, то такая дата встречается только в отдельных точках, так что наверное это из-за неё.
```
		<trkseg>
			<trkpt lat="60.053317" lon="30.446924"><time>1970-01-01T00:00:01.000Z</time></trkpt>
			<trkpt lat="60.049648" lon="30.444617"><time>1970-01-01T00:00:01.000Z</time></trkpt>
			<trkpt lat="60.049466" lon="30.444349"><time>1970-01-01T00:00:01.000Z</time></trkpt>

```

Из тикета https://github.com/wladich/nakarte/issues/90 не понял, почему используется "левое" время, почему бы не ставить настоящее - время создания трэка или время экспорта?

**Comment by wladich** [comment]
Это вопрос совсем другой, чем оригиналный issue. Вынес в https://github.com/wladich/nakarte/issues/829

**Comment by balovp** [comment]
Вопрос всё еще актуален. После похода часто редактируем треки, удаляя "пауков" на сегментах (скопления точек, когда остановились на привал или топтались на одном месте ради фотографий). Раньше пользовались GPS Track Editor http://www.gpstrackeditor.com/ но на него уже больше 8 лет нет обновлений. Похожая функция есть во многих других веб-сервисах (например, в manymaps, но они полу-платные), но она обычно криво реализована. 
В nakarte же удобно удалять пауков, но тогда сбрасывается время в точках. 

Если есть возможность при чтении треков сохранять время и высоту, а при выгрузке добавить диалоговое окно с выбором "адаптировать трек для Garmin (т.е. удалить время, высоту и температуру) из данных трека" или "не адаптировать под Garmin (сохранить время, высоту и температуру)", то это бы очень помогло нам!
